### PR TITLE
add trim function, add ut for trim funcions

### PR DESCRIPTION
**Pull request by QuantumBear**
I hereby agree to the terms of the CLA available at: https://databend.rs/policies/cla/

## Summary

Summary about this PR

## Changelog

- New Feature

## Related Issues

Fixes #issue

## Test Plan

Unit Tests

Stateless Tests

### CODE REVIEW

**Review by sundy-li — APPROVED** [review]

**Comment by zhyass** [comment]
Hi, @QuantumBear 
you can run `cargo clippy --tests -- -D warnings` and `cargo fmt` before commit.

**Comment by BohuTANG** [comment]
> Hi, @QuantumBear
> you can run `cargo clippy --tests -- -D warnings` and `cargo fmt` before commit.

We have a short command: make lint

File: common/functions/src/scalars/strings/trim.rs [hunk 1]
```diff
@@ -43,5 +43,19 @@ impl StringOperator for RTrim {
     }
 }
 
+#[derive(Clone, Default)]
+pub struct Trim;
+
+impl StringOperator for Trim {
+    fn apply<'a>(&'a mut self, s: &'a [u8]) -> Option<&'a [u8]> {
```
Comment (sundy-li) [inline common/functions/src/scalars/strings/trim.rs:50]: `#[inline]` is better

Comment (QuantumBear) [reply common/functions/src/scalars/strings/trim.rs:50]: Thanks, I'll update it this evening.

**Pull request merged**